src/connector/src/sink/mongodb.rs

```diff
@@ -395,49 +387,42 @@ impl MongodbSinkWriter {
                 }
                 continue;
             }
-            self.payload_writer.append(insert_builder, row).await?;
+            self.payload_writer.append(&mut insert_builder, row)?;
         }
-        Ok(())
+        self.payload_writer.flush_insert(&mut insert_builder)
```

Comment:
The `flush_insert` can change to consume the ownership of `insert_builder`. So can `flush_upsert`.